src/icp_canister_backend/tests/rewards_tests.rs

```diff
@@ -159,4 +196,672 @@ fn test_reward_distribution_middle_and_final() {
+        ledger_id,
+        large_depositor,
+        large_deposit,
+        episode,
+    );
+    create_deposit(
+        &pic,
+        canister_id,
+        ledger_id,
+        small_depositor,
+        small_deposit,
+        episode,
+    );
+
+    let reward_start_time = pic.get_time().as_nanos_since_unix_epoch() / 1_000_000_000;
+    reward_pool(
+        &pic,
+        canister_id,
+        ledger_id,
+        large_depositor,
+        reward_amount.clone(),
+    )
+    .expect("Reward pool should succeed");
+
+    let last_reward_episode = (reward_start_time + EPISODE_DURATION * 12) / EPISODE_DURATION;
+    let exact_reward_duration = (last_reward_episode + 1) * EPISODE_DURATION - reward_start_time;
+    advance_time(&pic, exact_reward_duration);
+
+    let large_depositor_rewards = pic
+        .query_call(
+            canister_id,
+            large_depositor,
+            "get_deposits_rewards",
+            encode_args((vec![0u64],)).unwrap(),
+        )
+        .map(|r| decode_one::<Nat>(&r).unwrap())
+        .expect("Failed to get large depositor rewards");
+
+    let small_depositor_rewards = pic
+        .query_call(
+            canister_id,
+            small_depositor,
+            "get_deposits_rewards",
+            encode_args((vec![1u64],)).unwrap(),
+        )
+        .map(|r| decode_one::<Nat>(&r).unwrap())
+        .expect("Failed to get small depositor rewards");
+
+    //Get actual shares to compute precise expected values
+    let large_deposits: Vec<icp_canister_backend::UserDepositInfo> = decode_one(
+        &pic.query_call(
+            canister_id,
+            large_depositor,
+            "get_user_deposits",
+            encode_args((large_depositor,)).unwrap(),
+        )
+        .expect("Failed to get large depositor info"),
+    )
+    .unwrap();
+    let small_deposits: Vec<icp_canister_backend::UserDepositInfo> = decode_one(
+        &pic.query_call(
+            canister_id,
+            small_depositor,
+            "get_user_deposits",
+            encode_args((small_depositor,)).unwrap(),
+        )
+        .expect("Failed to get small depositor info"),
+    )
+    .unwrap();
+
+    let large_shares = &large_deposits[0].shares;
+    let small_shares = &small_deposits[0].shares;
+    let total_shares = large_shares.clone() + small_shares.clone();
+
+    let expected_large = (reward_amount.clone() * large_shares.clone()) / total_shares.clone();
+    let expected_small = (reward_amount.clone() * small_shares.clone()) / total_shares.clone();
+
+    let one_sat_allowed_error = Nat::from(1u64);
+    assert_with_error!(
+        &large_depositor_rewards,
+        &expected_large,
+        &one_sat_allowed_error,
+        "Large depositor rewards"
+    );
+    assert_with_error!(
+        &small_depositor_rewards,
+        &expected_small,
+        &one_sat_allowed_error,
+        "Small depositor rewards"
+    );
+
+    let total_distributed = large_depositor_rewards + small_depositor_rewards;
```

Comment:
put total_distributed upper
And finish the test with 3 asserts
By the way what is the expected_small?
If it is 1, then you should either use exact assert because we should not allow a situation when  small_depositor_rewards is 0.